src/nn/ops/conv_transpose2d.ts

```diff
@@ -160,6 +159,68 @@ export class ConvTranspose2d extends SingleOutputOperation
     return inputs;
   }
 
+  createOutput(): void {
+    let batches, channels, inputHeight, inputWidth, filterHeight, filterWidth;
+    this.tfPadding_ = utils.getPaddings(this.padding_);
+    const inputShape = this.input_.shape();
+
+    const filterShape = this.filter_.shape();
+    if (this.filterLayout_ === MLConvTranspose2dFilterOperandLayout.iohw) {
+      channels = filterShape[1];
+      filterHeight = filterShape[2];
+      filterWidth = filterShape[3];
+    } else if (
+        this.filterLayout_ === MLConvTranspose2dFilterOperandLayout.hwoi) {
+      filterHeight = filterShape[0];
+      filterWidth = filterShape[1];
+      channels = filterShape[2];
+    } else if (
+        this.filterLayout_ === MLConvTranspose2dFilterOperandLayout.ohwi) {
+      channels = filterShape[0];
+      filterHeight = filterShape[1];
+      filterWidth = filterShape[2];
+    }
+
+    if (this.inputLayout_ === MLInputOperandLayout.nchw) {
+      [batches, , inputHeight, inputWidth] = inputShape;
+      this.outputShape_ = [batches, channels, 0, 0];
+    } else {
+      // nhwc layout
+      [batches, inputHeight, inputWidth, ] = inputShape;
+      // tf.convTranspose2d outputShape: [batch, height, width, outDepth]
```

Comment:
Why mention tf.convTranspose2d here? The shape inference code should be backend agnostic.